### PR TITLE
DHFPROD-6549: Explorer shows datahub artifacts when All Entities is selected

### PR DESCRIPTION
**Pull request by xnikhil08**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: marklogic-data-hub-central/ui/src/pages/Browse.tsx [hunk 1]
```diff
@@ -148,13 +148,19 @@ const Browse: React.FC<Props> = ({location}) => {
     if (searchOptions.nextEntityType && searchOptions.nextEntityType !== "All Data") {
       setCardView(false);
     }
-    if ((entities.length || (searchOptions.nextEntityType === "All Data" || searchOptions.nextEntityType === "All Entities" || searchOptions.nextEntityType === undefined)) &&
-      (!searchOptions.nextEntityType ||
-        (searchOptions.nextEntityType === "All Entities" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && !cardView) ||
+    if ((entities.length > 0 || (searchOptions.nextEntityType === "All Data" || searchOptions.nextEntityType === "All Entities" || searchOptions.nextEntityType === undefined)) &&
```
Comment (rjrudin) [inline marklogic-data-hub-central/ui/src/pages/Browse.tsx:151]: Something to consider here - there are a lot of conditionals here, making it hard for a human to read this and understand what the conditions are. 

The Clean Code book recommends defining local variables to capture the names of the conditions here, which then makes the overall conditional much easier to read. I think that would help a lot here. 

For example, the first condition above is saying - if at least one entity type exists (I am assuming that "entities" is really "entityTypes" or the user has selected All Data or All Entities or has not selected anything (and I'm assuming that "nextEntityType" is really "what the user selected" - "next" is confusing here). So how about:

    const entityTypesExistOrNoEntityTypeIsSelected = (all the above logic)

Reading that... I'm wondering when that would ever be false?? Maybe "nextEntityType" isn't really "what the user has selected in the search type dropdown"? I'm confused.

Regardless, It looks like all of the logic can then be simplified to something that reads like this:

    if (entityTypesExistOrNoEntityTypeIsSelected && (nameForFirstCondition || nameForSecondCondition || nameForThirdCondition || nameForFourthCondition))

Which would be far easier to understand than the current code.

File: marklogic-data-hub-central/ui/src/pages/Browse.tsx [hunk 2]
```diff
@@ -148,13 +148,19 @@ const Browse: React.FC<Props> = ({location}) => {
...
+      ((!searchOptions.nextEntityType && entities.length > 0) ||
+        (searchOptions.nextEntityType === "All Entities" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && !cardView && entities.length > 0) ||
         (searchOptions.nextEntityType === "All Data" && !searchOptions.entityTypeIds.length && !searchOptions.selectedTableProperties.length && cardView) ||
-        (!["All Entities", "All Data"].includes(searchOptions.nextEntityType) && searchOptions.entityTypeIds[0] === searchOptions.nextEntityType)
+        (searchOptions.nextEntityType && !["All Entities", "All Data"].includes(searchOptions.nextEntityType) && searchOptions.entityTypeIds[0] === searchOptions.nextEntityType)
       )) {
       getSearchResults(entities);
+    } else {
+      if (!cardView && (!searchOptions.entityTypeIds.length || !searchOptions.nextEntityType)) {
```
Comment (timur-isangulov) [inline marklogic-data-hub-central/ui/src/pages/Browse.tsx:159]: This condition crashes the app due to setting facet object to be empty if apply facets and refresh the page or open detail page and return back to browse. 
Screen Shot 2021-02-09 at 8 29 08 AM

**Review by timur-isangulov — CHANGES_REQUESTED** [review]
There is no search request call if select All Data option on Explorer zero state page. 
https://drive.google.com/file/d/1IS5TDhyxV9lz9ES-h0HH-K3R9EqWJT6D/view

File: marklogic-data-hub-central/ui/src/pages/Browse.tsx (hunk 1 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/pages/Browse.tsx]: Refactored the code a bit to better understand the logic.

File: marklogic-data-hub-central/ui/src/pages/Browse.tsx (hunk 2 above)
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/pages/Browse.tsx]: This is resolved.

**Comment by xnikhil08** [comment]
> There is no entity search request call if select All Data option on Explorer zero state page.
> https://drive.google.com/file/d/1IS5TDhyxV9lz9ES-h0HH-K3R9EqWJT6D/view

@timur-isangulov - This is already working fine for me. Can you please check again now with updated refactoring ?

**Review by timur-isangulov — APPROVED** [review]
Checked out and tested, looks good.

**Review by rjrudin — APPROVED** [review]

**Comment by timur-isangulov** [comment]
@xnikhil08, one of the manage query test fails on pipeline, I can reproduce it manually. Can you please take a look.

**Review by rahulvudutala — APPROVED** [review]

**Comment by xnikhil08** [comment]
> @xnikhil08, one of the manage query test fails on pipeline, I can reproduce it manually. Can you please take a look.

@timur-isangulov - Pushed a fix. Another review ?

**Review by timur-isangulov — APPROVED** [review]

**Review by Sanjeevani19 — APPROVED** [review]

**Pull request merged**